cozumler/en-yakin-noktalar/cozum.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import math
+from math import sqrt
+
+coordinate = [(3, 1), (3, 4), (4, 2)]
+
+def distances(coordinate):
```

Comment:
Bu fonksiyon için şu kısımlar yanlış;
1 - `values` boş yere define edilmiş.
2 - PEP8'e uyulmayan noktalar (bknz `calculateValue`)
3 - math.pow kullanımı
4 - Performans düşürücü type conversionları
5 - Algortitma : listenin içindeki elemanların indexin alıp tek tek o elemanlara ulaşıp onları başka bir listeye aralarında bir boşluk olarak ekleyip alt tarafta bu listeye erişip eklenen elemanın boşluklarını ayırıp  tekrar bir listeye çevirip algoritmayı karıştırmak. 

Bunun çözümü;
1 - Bu fonksiyonda gereksiz yere 3 adet local liste define etmeye gerek yok. 
2 - Return yerine generator objesi döndürmek (yield ile) performansı arttıracak.
3 - For ile index içinde dolaşılacağına direk verilen listenin item'larının referansına ulaşmak çok daha efektif bir çözüm yolu.
4 - Parsingin kaldırılması
5 - math.pow yerine built-in pow kullanımı
6 - type conversion'ların azaltılması

Senin koduna bakarak aynı inputtan aynı outputu döndüren şu kodu yazdım; 
```py
def distances(coords):
    for coord in coords:
         yield sqrt(pow(coord[0], 2) + pow(coord[1], 2))
```